### ISSUE TITLE
Crashes on first run with illegal argument exception

### ISSUE THREAD

**Issue by citrusella** [enteraname74/SoulSearching#36]
I have a TCL 10 5G UW running Android 10 and wanted to give this app a shot. On first run it requests file permission. I grant it, it starts trying to search songs and gets about 5/6 of the way through the little loading bar... and crashes. All subsequent crashes are nearly immediate, but when they load long enough for me to tell what screen they're on, it's the searching-for-songs screen.

Here's the logcat entry:

```
FATAL EXCEPTION: main
Process: com.github.enteraname74.soulsearching, PID: 17969
java.lang.IllegalArgumentException: Cannot coerce value to an empty range: maximum -1.57148672E9 is less than minimum 0.0.
	at S4.d.y(Unknown Source:44)
	at O.q2.c(Unknown Source:6)
	at O.o2.a(Unknown Source:317)
	at K5.C.e(Unknown Source:630)
	at F3.l.j(Unknown Source:1486)
	at Z.a.j(Unknown Source:51)
	at R.q.a(Unknown Source:221)
	at F3.m.a(Unknown Source:966)
	at P3.h.i(Unknown Source:215)
	at Z.a.e(Unknown Source:43)
	at Z.a.i(Unknown Source:8)
	at B.F.j(Unknown Source:111)
	at Z.a.j(Unknown Source:51)
	at B.F.j(Unknown Source:171)
	at Z.a.j(Unknown Source:51)
	at R.q.I(Unknown Source:14)
	at R.p.r(Unknown Source:121)
	at R.p.k(Unknown Source:10)
	at R.u.k(Unknown Source:35)
	at R.u0.a(Unknown Source:44)
	at R.n.a(Unknown Source:4)
	at R.u.l(Unknown Source:8)
	at v0.B.i(Unknown Source:21)
	at v0.B.h(Unknown Source:118)
	at v0.w.X(Unknown Source:170)
	at B.F.j(Unknown Source:65)
	at v0.y.d(Unknown Source:77)
	at x0.t.d(Unknown Source:37)
	at y.x.d(Unknown Source:96)
	at x0.y.d(Unknown Source:10)
	at h0.d.b(Unknown Source:50)
	at b0.r.d(Unknown Source:59)
	at b0.u.a(Unknown Source:47)
	at b0.v.c(Unknown Source:152)
	at x0.d0.a(Unknown Source:2)
	at x0.H.z0(Unknown Source:128)
	at x0.H.d(Unknown Source:95)
	at y.O.d(Unknown Source:185)
	at x0.t.d(Unknown Source:37)
	at h0.d.b(Unknown Source:50)
	at b0.r.d(Unknown Source:59)
	at b0.u.a(Unknown Source:47)
	at b0.v.c(Unknown Source:152)
	at x0.d0.a(Unknown Source:2)
	at x0.H.z0(Unknown Source:128)
	at x0.H.d(Unknown Source:95)
	at y.o.d(Unknown Source:82)
	at x0.t.d(Unknown Source:37)
	at y.x.d(Unknown Source:96)
	at x0.y.d(Unknown Source:10)
	at h0.d.b(Unknown Source:50)
	at b0.r.d(Unknown Source:59)
	at b0.u.a(Unknown Source:47)
	at b0.v.c(Unknown Source:152)
	at x0.d0.a(Unknown Source:2)
	at x0.H.z0(Unknown Source:128)
	at x0.H.d(Unknown Source:95)
	at v0.Q.d(Unknown Source:38)
	at x0.t.d(Unknown Source:37)
	at h0.d.b(Unknown Source:50)
	at b0.r.d(Unknown Source:59)
	at b0.u.a(Unknown Source:47)
	at b0.v.c(Unknown Source:152)
	at x0.d0.a(Unknown Source:2)
	at x0.H.z0(Unknown Source:128)
	at x0.O.d(Unknown Source:17)
	at x0.O.n(Unknown Source:17)
	at x0.O.k(Unknown Source:62)
	at y0.r.onMeasure(Unknown Source:82)
	at android.view.View.measure(View.java:25090)
	at y0.a.f(Unknown Source:63)
	at y0.a.onMeasure(Unknown Source:3)
	at android.view.View.measure(View.java:25090)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6872)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
	at androidx.appcompat.widget.ContentFrameLayout.onMeasure(Unknown Source:159)
	at android.view.View.measure(View.java:25090)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6872)
	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
	at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
	at android.view.View.measure(View.java:25090)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6872)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
	at android.view.View.measure(View.java:25090)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6872)
	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
	at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
	at android.view.View.measure(View.java:25090)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6872)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
	at com.android.internal.policy.DecorView.onMeasure(DecorView.java:742)
	at android.view.View.measure(View.java:25090)
	at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:3081)
	at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1855)
```

When I started writing this issue, I had 0.8.0 downloaded from IzzyOnDroid via F-Droid. In the interest of making sure it wasn't already fixed, I downloaded 0.8.1 directly from Github and tried it; it crashes with the same behavior and error.

**Comment by enteraname74** [comment]
Thanks for the feedback!

So, without doing anything, while the app search for songs on your device, it leads to a crash. So the crash always occurs on the “search for songs” screen and the app can never finish the fetch of the songs? Do the crash always occurs when the progress bar reaches 5/6 of its size, or is this really just after you open the app after the first crash?

Maybe it's the progress bar itself that may be managed incorrectly, leading to potential crashes. 
I will investigate this and try to find why this crash occurs. If you have any other information that could help to resolve this, feel free to share it. 
Anyway, I will let you know if I found something.

**Comment by citrusella** [comment]
Yes, it seems to always be during search for songs and it doesn't seem to finish. It's only 5/6 of the way through the bar the very first time, a handful of seconds after permission has been given (I've tested the "first" crash a few times by way of clearing the app's data and trying again to see if anything would be different). All subsequent crashes come faster--either the app has just barely started to display the searching and crashes (so after a second or so) or it crashes before the app can show *anything* (immediately).

I have less than 100 songs on my phone, probably less than 50. (Another app I just looked at has 44 files in its list.)

I can't think of anything else important, unless maybe it's relevant that my phone doesn't have an SD card in it. Everything's just on the internal memory.

**Comment by enteraname74** [comment]
Thanks for those informations.
I was thinking that it may have been a performance issue with lots of songs but it seems to not be the case if you have less than 100 songs. 
It may just be a UI issue with the progress bar that somehow uses wrong values to increment itself. 
I'm working on app performance currently. I will try to find what causes this crash and hopefully fix this for the next release. I will let you know when a new release is out 👍

**Comment by enteraname74** [comment]
Normally, this bug is now fixed in the latest release of the app (v0.9.0). 
I encourage you to do a fresh installation with this new release. Let me know if it works for you :+1:

**Comment by citrusella** [comment]
I was about to update and noticed that for whatever reason the bug stopped happening on 0.8.1. (I still intend to update but wanted to share that update before I did so...
